subt/main.py

```diff
@@ -479,8 +483,6 @@ def on_pose3d(self, timestamp, data):
         xyz, rot = data
         ypr = quaternion.euler_zyx(rot)
         x0, y0, z0 = self.offset
-        # pretend that received coordinates are in robot frame (compatible with on_pose2d)
-        xyz = xyz[0] - x0, xyz[1] - y0, xyz[2] - z0
```

Comment:
Why it is now ok to use the global coordinates as is? If it is ok, we don't need the offset at all and then the addition of `if self.offset is None` is not needed either. I am confused.

Reply:
+1, this needs to be fixed. The ROS odometry is zero-based and does not already contain the offset, like old Visual Odometry did. We should not subtract the offset on this line anymore and we should continue taking it into account (i.e. adding it) where needed.